### ISSUE TITLE
Improve Documentation

### ISSUE THREAD

**Issue by zachgk** [awslabs/djl#44]
The documentation in DJL was originally written with the expectation that users are reasonably familiar with deep learning. So, it does not go out of the way to define and explain some of the key concepts. To help users who are newer to deep learning, we created a [documentation convention](https://github.com/awslabs/djl/blob/master/docs/development/development_guideline.md#documentation-conventions) for what explanation is required to get a basic understanding of the relevant topics. We now need to update the existing javadocs to contain all the required information.

- [ ] Blocks
  - [ ] Convolution (and Conv1D, Conv2D, Conv3D)
  - [ ] Embedding Block
  - [ ] Linear Block
  - [ ] Activations (mostly in `Activation`)
    - [ ] Prelu block
  - [ ] BatchNorm
  - [ ] Dropout
  - [ ] Pooling (in `Pool`)
  - [ ] GRU
  - [ ] LSTM
  - [ ] RNN
- [ ] Evaluators
- [ ] Loss Functions
- [ ] Initializers
- [ ] Optimizers
- [ ] Models
  - [ ] Mlp
  - [ ] ResnetV1
  - [ ] SingleShotDetection
- [ ] Datasets
  - [ ] Captcha
  - [ ] Cifar10
  - [ ] Coco
  - [ ] ImageNet
  - [ ] Mnist
  - [ ] PikachuDetection
  - [ ] StanfordMovieReview
  - [ ] TatoebaEnglishFreshDataset

Comment below if you are interested in helping with any of the documentation or if you notice any other javadoc that does not match the convention. I will edit this description to keep it up to date as the documentation is updated.

**Comment by jonathan016** [comment]
Hi @zachgk , as we've discussed on Slack, I'd like to help on improving documentation on the following javadocs:
- [ ] Convolution (and Conv1D, Conv2D, Conv3D)
- [ ] BatchNorm
- [ ] Dropout

The next steps would be forking the project and improving the documentation in accordance with the [documentation convention](https://github.com/awslabs/djl/blob/master/docs/development/development_guideline.md#documentation-conventions), right? Thank you in advance.

**Comment by zachgk** [comment]
Thanks @jonathan016. I've added you up above. You'll just want to fork, update to the convention you linked, and then open a PR with your improvement.
